### ISSUE TITLE
[Precision] bf16 nn.Linear backward weight gradient mismatch between CPU and XPU between nightly xpu 0301/0303

### ISSUE THREAD

**Issue by Chao1Han** [intel/torch-xpu-ops#3023]
### 🐛 Describe the bug

Reproduce: pytest -sv test...
```python
import torch
import torch.nn as nn
import torch.nn.functional as F
from torch.testing._internal.common_utils import TestCase


class TestNNMethod(TestCase):
    
    def test_linear_bfloat16(self, dtype=torch.bfloat16):
        linear = nn.Linear(4, 2, dtype=dtype)
        tanh = nn.Tanh()
        x_cpu = torch.tensor(
            [[1.23, 2.34, 6.45, 2.22], [0.23, 1.34, 7.45, 1.22]],
            requires_grad=True,
            dtype=dtype,
        )
        z_cpu = linear(x_cpu)
        y_cpu = tanh(z_cpu)
        y_cpu.backward(torch.tensor([[1.01, 8.32], [2.4, 3.22]]))
        linear_weight_grad_cpu = linear.weight.grad.clone()
        linear.zero_grad()

        linear_dpcpp = linear.to("xpu")
        tanh.to("xpu")

        x_dpcpp = torch.tensor(
            [[1.23, 2.34, 6.45, 2.22], [0.23, 1.34, 7.45, 1.22]],
            requires_grad=True,
            device="xpu",
            dtype=dtype,
        )

        z_dpcpp = linear_dpcpp(x_dpcpp)
        y_dpcpp = tanh(z_dpcpp)

        y_dpcpp.backward(torch.tensor([[1.01, 8.32], [2.4, 3.22]], device="xpu"))
        linear_weight_grad_dpcpp = linear_dpcpp.weight.grad.clone()

        self.assertEqual(linear_weight_grad_cpu, linear_weight_grad_dpcpp)

```
maybe https://github.com/pytorch/pytorch/pull/175988/changes expose accuracy gap.
### Versions

2.12.0.dev20260301+xpu pass (git commit 434b8cfdfac8ef7c741385561e340781ad76512f)
2.12.0.dev20260303+xpu fail (git commit 69725aae8d299c39c8b43ef6282129ddfebd45dc)

**Comment by CuiYifeng** [comment]
@Chao1Han Please verify, thanks.
